platforms/core-configuration/file-collections/src/main/java/org/gradle/api/internal/file/DefaultFilePropertyFactory.java

```diff
@@ -180,10 +181,13 @@ public RegularFile transform(CharSequence path) {
         }
     }
 
-    private static abstract class AbstractFileVar<T extends FileSystemLocation, THIS extends FileSystemLocationProperty<T>> extends DefaultProperty<T> implements FileSystemLocationProperty<T> {
+    private static abstract class AbstractFileVar<T extends FileSystemLocation, THIS extends FileSystemLocationProperty<T>> extends DefaultProperty<T> implements FileSystemLocationProperty<T>, ConventionMappingFileSystemLocationPropertyProxy {
+
+        private final Class<T> type;
```

Comment:
Is this an optimization? (given that the type is available via `DefaultProperty.getType()`)

Reply:
No, it's not an optiomization, I just didn't know that `getType()` exists. Thanks for the insight, I will use that!